lib/service/cfg.go

```diff
@@ -1001,8 +1007,13 @@ func (d *Database) CheckAndSetDefaults() error {
 		return trace.BadParameter("missing Cloud SQL project ID for database %q", d.Name)
 	}
 
-	// For SQL Server we only support Kerberos auth with Active Directory at the moment.
-	if d.Protocol == defaults.ProtocolSQLServer {
+	// We support Azure AD authentication and Kerberos auth with AD for SQL
+	// Server. The first method doesn't require additional configuration since
+	// it assumes the environment’s Azure credentials
+	// (https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication).
+	// The second method requires additional information, validated by
+	// DatabaseAD.
+	if d.Protocol == defaults.ProtocolSQLServer && (!d.AD.IsEmpty() || !strings.Contains(d.URI, azureutils.MSSQLEndpointSuffix)) {
```

Comment:
Why do we check if the URI contains the suffix here?

Reply:
To know if we should validate the AD configuration or not. This check is the same as in the connection flow.